cmd/tlssocks/tlssocks.go

```diff
@@ -25,53 +25,57 @@ func init() {
 }
 
 type socksAuthenticator struct {
-	Destinations  map[string]*Destination
+	Destinations map[string]*Destination
+
 	resolvedNames map[string][]string
 }
 
-func newSocksAuthenticator(destinations map[string]*Destination) (suxx5 *socksAuthenticator, err error) {
-	suxx5 = &socksAuthenticator{
+func newSocksAuthenticator(destinations map[string]*Destination) (*socksAuthenticator, error) {
+	suxa := &socksAuthenticator{
 		Destinations: destinations,
 	}
-	names := []string{}
+	names := make([]string, 0, len(destinations))
 	for name := range destinations {
 		names = append(names, name)
 	}
-	resolvedNames, errResolveNames := resolveNames(names)
-	if errResolveNames != nil {
-		err = errResolveNames
-		return
+
+	resolvedNames, err := resolveNames(names)
+	if err != nil {
+		return nil, err
 	}
-	suxx5.resolvedNames = resolvedNames
+	suxa.resolvedNames = resolvedNames
+
 	go func() {
 		time.Sleep(time.Second * 10)
-		resolvedNames, errResolveNames := resolveNames(names)
-		if errResolveNames == nil {
-			suxx5.resolvedNames = resolvedNames
+
+		resolvedNames, err := resolveNames(names)
+		if err == nil {
+			suxa.resolvedNames = resolvedNames
 		} else {
-			logger.Warn("could not resolve names: " + errResolveNames.Error())
+			logger.Warn("could not resolve names: " + err.Error())
 		}
 	}()
-	return
+	return suxa, nil
 }
 
-func resolveNames(names []string) (newResolvedNames map[string][]string, err error) {
-	newResolvedNames = map[string][]string{}
+func resolveNames(names []string) (map[string][]string, error) {
+	newResolvedNames := map[string][]string{}
 	for _, name := range names {
-		addrs, errLookup := net.LookupHost(name)
-		if errLookup != nil {
-			err = errLookup
-			return
+		addrs, err := net.LookupHost(name)
+		if err != nil {
+			return nil, err
```

Comment:
This might be odd, given that if a DNS entry changes for any given reason, this will still hold the cached version THUS not respecting the DNS TTL

Reply:
We can think about that when this really happens ... how often does this occur?
A quick look in the stdlib code ... there is no possibility for a TTL but we can go down this route: https://github.com/miekg/dns